src/OpenTelemetry.Exporter.Prometheus/Implementation/PrometheusSerializerExt.cs

```diff
@@ -93,7 +96,7 @@ public static int WriteMetric(byte[] buffer, int cursor, Metric metric)
                     var bucketCounts = metricPoint.BucketCounts;
                     var explicitBounds = metricPoint.ExplicitBounds;
                     long totalCount = 0;
-                    for (int idxBound = 0; idxBound < explicitBounds.Length + 1; idxBound++)
+                    for (int idxBound = 0; idxBound < (explicitBounds != null ? explicitBounds.Length + 1 : 0); idxBound++)
```

Comment:
nit: Might be a few nanos faster to wrap the `for` in an outer `if (explicitBounds != null)` so it doesn't have to do the length evaluation each loop. Assuming that isn't optimized away somehow.